src/Confluent.Kafka/Impl/SafeKafkaHandle.cs

```diff
@@ -258,9 +258,9 @@ internal WatermarkOffsets GetWatermarkOffsets(string topic, int partition)
             return new WatermarkOffsets(low, high);
         }
 
-        internal void Subscribe(ICollection<string> topics)
+        internal void Subscribe(IEnumerable<string> topics)
         {
-            IntPtr list = LibRdKafka.topic_partition_list_new((IntPtr) topics.Count);
+            IntPtr list = LibRdKafka.topic_partition_list_new((IntPtr) topics.Count());
```

Comment:
In reply to my own comment above, the LINQ `Count()` implementation is evidently fast if the collection is an `ICollection`: http://stackoverflow.com/questions/168901/howto-count-the-items-from-a-ienumerablet-without-iterating

... but now I'm wondering under what circumstances anyone would ever want to use a data structure that doesn't implement `ICollection` in these methods? @treziac has a good point - using `IEnumerable` opens up the possibility of an infinite loop. Seems unlikely any one would use it in this way, but if they're passing in something that isn't an `ICollection`, well, I think i would be inclined to worry about it a little bit.

So now i'm thinking that rather than changing from `ICollection` -> `IEnumerable`, we should do an audit of everywhere we're using `IEnumerable` and possibly think about changing to `ICollection`. 

Not that this is a big deal. Chances of anyone running into it are minimal.